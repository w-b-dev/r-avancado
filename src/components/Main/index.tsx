import React from "react";
const Main = ({ children }: { children?: JSX.Element }) => {
  return (
    <main>
      <h1>Hi, Main!</h1>
      {children}
    </main>
  );
};
export default Main;
