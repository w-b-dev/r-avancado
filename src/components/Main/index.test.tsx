import { render, screen } from "@testing-library/react";
import Main from "./index";

describe("<Main>", () => {
  it("should render correctly", () => {
    const { container } = render(<Main />);
    expect(container).toMatchSnapshot();
  });

  it("should render the text correctly", () => {
    const { container } = render(<Main />);
    const target = screen.getByRole("main");
    expect(target).toHaveTextContent(/Hi/i);
  });
});
