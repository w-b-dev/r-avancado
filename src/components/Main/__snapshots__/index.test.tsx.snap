// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main> should render correctly 1`] = `
<div>
  <main>
    <h1>
      Hi, Main!
    </h1>
  </main>
</div>
`;
